crates/bevy_core_pipeline/src/core_3d/mod.rs

```diff
@@ -179,37 +176,48 @@ impl Plugin for Core3dPlugin {
     }
 }
 
+/// Opaque 3D [`BinnedPhaseItem`]s.
 pub struct Opaque3d {
-    pub asset_id: AssetId<Mesh>,
-    pub pipeline: CachedRenderPipelineId,
-    pub entity: Entity,
-    pub draw_function: DrawFunctionId,
+    /// The key, which determines which can be batched.
+    pub key: Opaque3dBinKey,
+    /// An entity from which the mesh and other attributes will be fetched.
+    pub representative_entity: Entity,
+    /// The ranges of instances.
     pub batch_range: Range<u32>,
+    /// The dynamic offset.
```

Comment:
The dynamic offset needs to be part of the bin key as it breaks batches. But it isn’t known until the order of entities in the queue is known. If you run with WGPU_SETTINGS_PRIO=webgl2 then storage buffers will be disabled and the ‘batched uniform buffer’ will be used so you can test this case.

Reply:
Or, if it isn't part of the bin key, the batching would need to take into account the need to split batches every N instances, basically.

Reply:
I modified the batching to assign this properly.